cmd/vcert/commands.go

```diff
@@ -318,73 +346,127 @@ func doCommandGetcred1(c *cli.Context) error {
 		return fmt.Errorf("could not create TPP connector: %s", err)
 	}
 
-	if cfg.Credentials.RefreshToken != "" {
-		resp, err := tppConnector.RefreshAccessToken(&endpoint.Authentication{
-			RefreshToken: cfg.Credentials.RefreshToken,
-			ClientId:     flags.clientId,
-			Scope:        flags.scope,
-		})
-		if err != nil {
-			return err
+	switch c.Command.Name {
+	case commandGetCredName:
+		//TODO: quick workaround to supress logs when output is in JSON.
+		if flags.credFormat != "json" {
+			logf("Getting credentials...")
 		}
-		if flags.credFormat == "json" {
-			jsonData, err := json.MarshalIndent(resp, "", "    ")
+
+		if cfg.Credentials.RefreshToken != "" {
+			resp, err := tppConnector.RefreshAccessToken(&endpoint.Authentication{
+				RefreshToken: cfg.Credentials.RefreshToken,
+				ClientId:     flags.clientId,
+				Scope:        flags.scope,
+			})
+			if err != nil {
+				return err
+			}
+			if flags.credFormat == "json" {
+				jsonData, err := json.MarshalIndent(resp, "", "    ")
+				if err != nil {
+					return err
+				}
+				fmt.Println(string(jsonData))
+			} else {
+				tm := time.Unix(int64(resp.Expires), 0).UTC().Format(time.RFC3339)
+				fmt.Println("access_token: ", resp.Access_token)
+				fmt.Println("access_token_expires: ", tm)
+				fmt.Println("refresh_token: ", resp.Refresh_token)
```

Comment:
is it not needed to do a nil validation like is in line 396 and 418?

Reply:
isnt line 356 the nil validation?

Reply:
that is right :-)